Sources/Rendering/OpenGL/CellArrayBufferObject/index.js

```diff
@@ -200,13 +232,71 @@ function vtkOpenGLCellArrayBufferObject(publicAPI, model) {
     let vboidx = 0;
     let ucidx = 0;
 
+    // Find out if shift scale should be used
+    let useShiftAndScale = false;
+    // Compute squares of diagonal size and distance from the origin
+    let diagSq = 0.0;
+    let distSq = 0.0;
+    for (let i = 0; i < 3; ++i) {
+      const range = options.points.getRange(i);
+
+      const delta = range[1] - range[0];
+      diagSq += delta * delta;
+
+      const distShift = 0.5 * (range[1] + range[0]);
+      distSq += distShift * distShift;
+    }
+    if (
+      diagSq > 0 &&
+      (Math.abs(distSq) / diagSq > 1.0e6 || // If data is far from the origin relative to its size
+      Math.abs(Math.log10(diagSq)) > 3.0 || // If the size is huge when not far from the origin
+        (diagSq === 0 && distSq > 1.0e6)) // If data is a point, but far from the origin
+    ) {
+      useShiftAndScale = true;
+    } else if (model.coordShiftAndScaleEnabled === true) {
+      // Make sure to reset
+      model.coordShiftAndScaleEnabled = false;
+      model.coordScale = null;
+      model.coordShift = null;
+    }
+
+    if (useShiftAndScale === true) {
+      // Compute shift and scale vectors
+      const shift = vec3.create();
+      const scale = vec3.create();
+      for (let i = 0; i < 3; ++i) {
+        const range = options.points.getRange(i);
+        const delta = range[1] - range[0];
+
+        shift[i] = 0.5 * (range[1] + range[0]);
+        scale[i] = delta > 0 ? 1.0 / delta : 1.0;
+      }
+
+      publicAPI.setCoordShift(shift);
+      publicAPI.setCoordScale(scale);
```

Comment:
it would seem to be more efficient to set both at once, create a `setCoordShiftAndScale` ?